lib/shoryuken/middleware/server/exponential_backoff_retry.rb

```diff
@@ -0,0 +1,54 @@
+module Shoryuken
+  module Middleware
+    module Server
+      class ExponentialBackoffRetry
+        include Util
+
+        def call(worker, queue, sqs_msg, body)
+          started_at = Time.now
+          yield
+        rescue Exception => e
+          retry_intervals = Array(worker.class.get_shoryuken_options['retry_intervals'])
+           
+          if retry_intervals.empty? || ! handle_failure(sqs_msg, started_at, retry_intervals)
+            # Re-raise the exception if the job is not going to be retried.
+            # This allows custom middleware (like exception notifiers) to be aware of the unhandled failure.
+            raise e
+          end
+        end
+        
+      private
+      
+        def handle_failure(sqs_msg, started_at, retry_intervals)
+          attempts = sqs_msg.attributes['ApproximateReceiveCount']
```

Comment:
How about `options[:attribute_names] ||= %[All]` so that users can still customize?  The middleware detects if the attribute is missing, so it should not cause a crash.